pkg/api/validation/validation.go

```diff
@@ -1278,6 +1291,18 @@ func ValidateSecret(secret *api.Secret) errs.ValidationErrorList {
 		}
 	case api.SecretTypeOpaque, "":
 		// no-op
+	case api.SecretTypeDockercfg:
+		dockercfgBytes, exists := secret.Data[api.DockerConfigKey]
+		if !exists {
+			allErrs = append(allErrs, errs.NewFieldRequired(fmt.Sprintf("data[%s]", api.DockerConfigKey)))
+			break
+		}
+
+		dockercfg := &credentialprovider.DockerConfig{}
+		if err := json.Unmarshal(dockercfgBytes, dockercfg); err != nil {
```

Comment:
Last comment before LGTM.

Why do we do this check here?

If it's because kubelet is going to need to parse & understand the .dockercfg file, then effectively this is part of the API and this struct definition needs to be versioned etc.

On the other hand, if it's just to be nice to the user & reject files that don't parse, then given the golang json parser, the exact struct you unmarshall into isn't relevant-- you could get the same verification my using a map[string]interface{} as the destination. But I question why we'd need to do this at all? Validating that it parses doesn't really give any assurance that it'll actually work with docker.